src/main/java/frc/robot/subsystems/drive/DriveSubsystem.java

```diff
@@ -4,51 +4,197 @@
+      DriveConstants.REAR_LEFT_TURNING_CAN_ID,
+      DriveConstants.BACK_LEFT_CHASSIS_ANGULAR_OFFSET);
+
+  private final SwerveModule m_rearRight = new SwerveModule(
+      DriveConstants.REAR_RIGHT_DRIVING_CAN_ID,
+      DriveConstants.REAR_RIGHT_TURNING_CAN_ID,
+      DriveConstants.BACK_RIGHT_CHASSIS_ANGULAR_OFFSET);
+
+  // The gyro sensor
+  private final ADIS16470_IMU m_gyro = new ADIS16470_IMU();
 
-  /** Creates a new ExampleSubsystem. */
-  public DriveSubsystem() {}
+  // Odometry class for tracking robot pose
+  SwerveDriveOdometry m_odometry = new SwerveDriveOdometry(
+      DriveConstants.DRIVE_KINEMATICS,
+      Rotation2d.fromDegrees(m_gyro.getAngle()),
+      new SwerveModulePosition[] {
+          m_frontLeft.getPosition(),
+          m_frontRight.getPosition(),
+          m_rearLeft.getPosition(),
+          m_rearRight.getPosition()
+      });
+
+  /** Creates a new DriveSubsystem. */
+  public DriveSubsystem() {
+  }
+
+  @Override
+  public void periodic() {
+    // Update the odometry in the periodic block
+    m_odometry.update(
+        Rotation2d.fromDegrees(m_gyro.getAngle()),
+        new SwerveModulePosition[] {
+            m_frontLeft.getPosition(),
+            m_frontRight.getPosition(),
+            m_rearLeft.getPosition(),
+            m_rearRight.getPosition()
+        });
+  }
 
   /**
-   * Example command factory method.
+   * Returns the currently-estimated pose of the robot.
    *
-   * @return a command
+   * @return The pose.
    */
-  public CommandBase exampleMethodCommand() {
-    // Inline construction of command goes here.
-    // Subsystem::RunOnce implicitly requires `this` subsystem.
-    return runOnce(
-        () -> {
-          /* one-time action goes here */
-        });
+  public Pose2d getPose() {
+    return m_odometry.getPoseMeters();
   }
 
   /**
-   * An example method querying a boolean state of the subsystem (for example, a digital sensor).
+   * Resets the odometry to the specified pose.
    *
-   * @return value of some boolean subsystem state, such as a digital sensor.
+   * @param pose The pose to which to set the odometry.
    */
-  public boolean exampleCondition() {
-    // Query some boolean state, such as a digital sensor.
-    return false;
+  public void resetOdometry(Pose2d pose) {
+    m_odometry.resetPosition(
+        Rotation2d.fromDegrees(m_gyro.getAngle()),
+        new SwerveModulePosition[] {
+            m_frontLeft.getPosition(),
+            m_frontRight.getPosition(),
+            m_rearLeft.getPosition(),
+            m_rearRight.getPosition()
+        },
+        pose);
   }
 
-  @Override
-  public void periodic() {
-    // This method will be called once per scheduler run
+  /**
+   * Method to drive the robot using joystick info.
+   *
+   * @param xSpeed        Speed of the robot in the x direction (forward).
+   * @param ySpeed        Speed of the robot in the y direction (sideways).
+   * @param rot           Angular rate of the robot.
```

Comment:
Can we clarify this comments ot reflect that all of these are:
- desired speeds / rates rather than actual speeds and rates
- in [-1, 1]? Rather than in units like m/s or rad/s